keras_cv/models/segmentation/deeplab.py

```diff
@@ -180,12 +127,163 @@ def train_step(self, data):
         self.optimizer.minimize(total_loss, self.trainable_variables, tape=tape)
         return self.compute_metrics(images, y_true, y_pred, sample_weight=sample_weight)
 
+    def get_config(self):
+        return {
+            "classes": self.classes,
+            "backbone": self.backbone,
+            "spatial_pyramid_pooling": self.spatial_pyramid_pooling,
+            "segmentation_head": self.segmentation_head,
+            "segmentation_head_activation": self.segmentation_head_activation,
+        }
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class SegmentationHead(layers.Layer):
+    """Prediction head for the segmentation model
+
+    The head will take the output from decoder (eg FPN or ASPP), and produce a
+    segmentation mask (pixel level classifications) as the output for the model.
+
+    Args:
+        classes: int, the number of output classes for the prediction. This should
+            include all the classes (eg background) for the model to predict.
+        convs: int, the number of conv2D layers that are stacked before the final
+            classification layer. Default to 2.
+        filters: int, the number of filter/channels for the the conv2D layers. Default
+            to 256.
+        activations: str or 'tf.keras.activations', activation functions between the
+            conv2D layers and the final classification layer. Default to 'relu'
+        output_scale_factor: int, or a pair of ints, for upsample the output mask.
+            This is useful to scale the output mask back to same size as the input
+            image. When single int is provided, the mask will be scaled with same
+            ratio on both width and height. When a pair of ints are provided, they will
+            be parsed as (height_factor, width_factor). Default to None, which means
+            no resize will happen to the output mask tensor.
+        kernel_size: default 3; the kernel_size to be used in each of the `convs` blocks
+        use_bias: default False; whether to use bias or not in each of the `convs` blocks
+                Defaults to none since the blocks use `BatchNormalization` after each conv, rendering
+                bias obsolete
+        output_activation: default 'softmax', the activation to apply in the classification
+            layer (output of the head)
+
+    Sample code
+    ```python
+    # Mimic a FPN output dict
+    p3 = tf.ones([2, 32, 32, 3])
+    p4 = tf.ones([2, 16, 16, 3])
+    p5 = tf.ones([2, 8, 8, 3])
+    inputs = {3: p3, 4: p4, 5: p5}
+
+    head = SegmentationHead(classes=11)
+
+    output = head(inputs)
+    # output tensor has shape [2, 32, 32, 11]. It has the same resolution as the p3.
+    ```
+    """
+
+    def __init__(
+        self,
+        classes,
+        convs=2,
+        filters=256,
+        activations="relu",
+        output_scale_factor=None,
```

Comment:
I don't feel `output_scale_factor` is necessary for now, remove it?

Reply:
It's necessary in DeepLabV3+, which is next in line, and uses the same type of segmentation head.
Since this'll become proprietary for DLV3 and isn't reused, we can remove it.